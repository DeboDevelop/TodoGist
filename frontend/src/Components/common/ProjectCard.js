import React, { useState } from "react";
import "../../assets/ProjectCard.css";
import Modal from "react-bootstrap/Modal";
import axios from "axios";

function ProjectCard({ item, index, handleProject }) {
    const [show, setShow] = useState(false);
    const [title, setTitle] = useState("");
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const handleChange = value => {
        setTitle(value);
    };
    const updateProject = e => {
        e.preventDefault();
        let token = localStorage.getItem("token");
        axios
            .patch(`${process.env.REACT_APP_BACKEND_URL}/project/${item._id}`, {
                token: token,
                title: title,
            })
            .then(res => {
                setTitle("");
                setShow(false);
                handleProject(res.data, index);
            })
            .catch(err => console.log(err));
    };
    const openTodo = () => {
        console.log("Wait!!!!");
    };
    return (
        <>
            <div className="d-flex flex-row justify-content-between mx-4 my-2 p-1 card-div">
                <div>
                    <button className="btn green-button" onClick={handleShow}>
                        Edit
                    </button>
                    <span className="txt mx-4">{item.title}</span>
                </div>
                <div className="px-2">
                    <span className="txt px-2">{item.listOfTodo} Items</span>
                    <button className="btn green-button" onClick={() => openTodo()}>
                        Open Todo
                    </button>
                </div>
            </div>
            <Modal show={show} onHide={handleClose} animation={false}>
                <Modal.Header className="modal-body" closeButton>
                    <Modal.Title className="txt">Update Project Title</Modal.Title>
                </Modal.Header>
                <Modal.Body className="modal-body">
                    <form className="form-group">
                        <input
                            type="text"
                            className="form-control"
                            value={title}
                            onChange={e => handleChange(e.target.value)}
                        />
                        <button className="btn green-button mt-2" onClick={e => updateProject(e)}>
                            Update
                        </button>
                    </form>
                </Modal.Body>
                <Modal.Footer className="modal-body">
                    <button className="btn green-button" onClick={handleClose}>
                        Close
                    </button>
                </Modal.Footer>
            </Modal>
        </>
    );
}

export default ProjectCard;
