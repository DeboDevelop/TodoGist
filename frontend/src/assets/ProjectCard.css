.txt {
    color: #ffffff;
}

.card-div {
    background-color: #494848;
    border: solid;
    border-width: 1px;
    border-color: #000000;
    border-radius: 5px;
}

.modal-body {
    background-color: #494848;
}
