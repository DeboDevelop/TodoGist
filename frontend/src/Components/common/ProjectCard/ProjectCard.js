import React, { useState } from "react";
import "../../../assets/css/ProjectCard.css";
import Modal from "react-bootstrap/Modal";
import axios from "axios";
import { useHistory } from "react-router-dom";

function ProjectCard({ item, index, handleProject }) {
    let history = useHistory();
    const [show, setShow] = useState(false);
    const [title, setTitle] = useState("");
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const handleChange = value => {
        setTitle(value);
    };
    const updateProject = e => {
        e.preventDefault();
        let token = localStorage.getItem("token");
        axios
            .patch(`${process.env.REACT_APP_BACKEND_URL}/project/${item._id}`, {
                token: token,
                title: title,
            })
            .then(res => {
                setTitle("");
                setShow(false);
                handleProject(res.data, index);
            })
            .catch(err => console.log(err));
    };
    const openTodo = () => {
        history.push(`/project/${item._id}?name=${item.title}`);
    };
    return (
        <>
            <div className="d-flex flex-column justify-content-center align-items-center p-2 m-2 card-div">
                <span className="txt" data-testid="title">
                    {item.title}
                </span>
                <span className="txt" data-testid="listOfTodo">
                    {item.listOfTodo} Items
                </span>

                <div className="d-flex flex-row justify-content-center">
                    <button className="btn green-button mx-1" onClick={handleShow}>
                        Edit
                    </button>
                    <button className="btn green-button mx-1" onClick={() => openTodo()}>
                        Open Todo
                    </button>
                </div>
            </div>
            <Modal show={show} onHide={handleClose} animation={false}>
                <Modal.Header className="modal-body" closeButton>
                    <Modal.Title className="txt">Update Project Title</Modal.Title>
                </Modal.Header>
                <Modal.Body className="modal-body">
                    <form className="form-group">
                        <input
                            type="text"
                            className="form-control"
                            value={title}
                            onChange={e => handleChange(e.target.value)}
                        />
                        <button className="btn green-button mt-2" onClick={e => updateProject(e)}>
                            Update
                        </button>
                    </form>
                </Modal.Body>
                <Modal.Footer className="modal-body">
                    <button className="btn green-button" onClick={handleClose}>
                        Close
                    </button>
                </Modal.Footer>
            </Modal>
        </>
    );
}

export default ProjectCard;
