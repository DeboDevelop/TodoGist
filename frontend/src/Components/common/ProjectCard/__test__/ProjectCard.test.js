import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import ProjectCard from "../ProjectCard";

const item = {
    _id: "12345",
    title: "Testing",
    listOfTodo: 5,
};

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ProjectCard item={item} index={0} handleProject={null} />, div);
});

it("renders component correctly", () => {
    const { getByTestId } = render(<ProjectCard item={item} index={0} handleProject={null} />);
    expect(getByTestId("title")).toHaveTextContent("Testing");
    expect(getByTestId("listOfTodo")).toHaveTextContent(5);
});

it("matches snapshot", () => {
    const tree = renderer.create(<ProjectCard item={item} index={0} handleProject={null} />).toJSON();
    expect(tree).toMatchSnapshot();
});
