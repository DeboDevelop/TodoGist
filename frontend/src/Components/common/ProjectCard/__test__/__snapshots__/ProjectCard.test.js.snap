// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex flex-column justify-content-center align-items-center p-2 m-2 card-div"
>
  <span
    className="txt"
    data-testid="title"
  >
    Testing
  </span>
  <span
    className="txt"
    data-testid="listOfTodo"
  >
    5
     Items
  </span>
  <div
    className="d-flex flex-row justify-content-center"
  >
    <button
      className="btn green-button mx-1"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="btn green-button mx-1"
      onClick={[Function]}
    >
      Open Todo
    </button>
  </div>
</div>
`;
